rcl/src/rcl/arguments.c

```diff
@@ -1170,16 +1171,16 @@ _rcl_parse_external_log_config_file(
   char ** log_config_file)
 {
   RCL_CHECK_ARGUMENT_FOR_NULL(arg, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(log_config_file, RCL_RET_INVALID_ARGUMENT);
 
   const size_t param_prefix_len = sizeof(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE) - 1;
   if (strncmp(RCL_EXTERNAL_LOG_CONFIG_ARG_RULE, arg, param_prefix_len) == 0) {
     size_t outlen = strlen(arg) - param_prefix_len;
-    *log_config_file = allocator.allocate(sizeof(char) * (outlen + 1), allocator.state);
-    if (NULL == log_config_file) {
```

Comment:
This line is where the bug I fixed was, i.e. it should have been `if (NULL == *log_config_file) {`.